src/code/audio/audio_data.c

```diff
@@ -0,0 +1,895 @@
+#include "global.h"
+
+s16 sLowPassFilterData[16 * 8] = {
+    /* 0x0 */ 0,     0,     0,    32767, 0,    0,     0,     0, // Identity filter (delta function)
+    /* 0x1 */ 3854,  4188,  4398, 4469,  4398, 4188,  3854,  3416,
+    /* 0x2 */ 3415,  4314,  4915, 5126,  4915, 4314,  3415,  2351,
+    /* 0x3 */ 2636,  4433,  5762, 6252,  5762, 4433,  2636,  849,
+    /* 0x4 */ 1334,  4196,  6646, 7609,  6646, 4196,  1334,  -802,
+    /* 0x5 */ -265,  3421,  7292, 8944,  7292, 3421,  -265,  -1863,
+    /* 0x6 */ -1558, 2065,  7146, 9546,  7146, 2065,  -1558, -1682,
+    /* 0x7 */ -2353, 726,   7441, 11028, 7441, 726,   -2353, -697,
+    /* 0x8 */ -2252, -693,  7121, 11962, 7121, -693,  -2252, 668,
+    /* 0x9 */ -1373, -1819, 6299, 12298, 6299, -1819, -1373, 1484,
+    /* 0xA */ -213,  -2740, 5843, 13680, 5843, -2740, -213,  1494,
+    /* 0xB */ 980,   -3081, 4883, 14286, 4883, -3081, 980,   590,
+    /* 0xC */ 1769,  -2973, 3866, 14981, 3866, -2973, 1769,  -568,
+    /* 0xD */ 2023,  -2554, 2911, 16397, 2911, -2554, 2023,  -1391,
+    /* 0xE */ 1766,  -1918, 2016, 19800, 2016, -1918, 1766,  -1564,
+    /* 0xF */ 841,   -853,  863,  26829, 863,  -853,  841,   -820,
+};
+
+s16 sHighPassFilterData[15 * 8] = {
+    /* 0x0 */ -289,  -291,  -289,  30736, -289,  -291,  -289,  -290,
+    /* 0x1 */ -464,  -467,  -467,  29506, -467,  -467,  -464,  -463,
+    /* 0x2 */ -662,  -670,  -672,  28101, -672,  -670,  -662,  -656,
+    /* 0x3 */ -839,  -855,  -861,  26830, -861,  -855,  -839,  -822,
+    /* 0x4 */ -996,  -1024, -1038, 25685, -1038, -1024, -996,  -963,
+    /* 0x5 */ -1184, -1236, -1266, 24272, -1266, -1236, -1184, -1118,
+    /* 0x6 */ -1357, -1450, -1506, 22900, -1506, -1450, -1357, -1238,
+    /* 0x7 */ -1514, -1680, -1784, 21498, -1784, -1680, -1514, -1307,
+    /* 0x8 */ -1613, -1877, -2048, 20390, -2048, -1877, -1613, -1298,
+    /* 0x9 */ -1657, -2185, -2559, 18869, -2559, -2185, -1657, -1093,
+    /* 0xA */ -1524, -2395, -3078, 18030, -3078, -2395, -1524, -739,
+    /* 0xB */ -1253, -2504, -3621, 17642, -3621, -2504, -1253, -367,
+    /* 0xC */ -525,  -2367, -4732, 17517, -4732, -2367, -525,  0,
+    /* 0xD */ -34,   -1762, -5706, 17503, -5706, -1762, -34,   -258,
+    /* 0xE */ -772,  -3,    -6985, 17240, -6985, -3,    -772,  -3,
+};
+
+s16 D_801D3070[105 * 8] = {
```

Comment:
Any idea what this data is?

Reply:
Yes, both `D_801D3070` and `D_801D3700` are filtering data that's used when both a lowPassCutoff and a highPassCutoff is applied.

`D_801D3070` is used specifically when `lowPassCutoff < highPassCutoff`

`D_801D3700` is used specifically when `lowPassCutoff > highPassCutoff`

I've left then unnamed as I still intent to do a proper fourier analysis on this data to understand what type of band-pass filter these were designed to be, and want to wait until that analysis to name them

Reply:
Edit: I'll also add that it's a bug when `lowPassCutoff = highPassCutoff` (one of the many bugs in this function that handles this data).

A different bug also forgot to multiply by 8, meaning this data is basically entirely used wrong. One instance of this data being used is by Guru Guru in Stock Pot Inn, but specifically when you are not in the same room as them, as you can here Guru Guru from the hallway if you move close enough (there's a separate bug in audio that doesn't initialize a variable properly, so you need to hear guru guru somewhere else first before the audio plays correctly)

Reply:
> 

Can we convert this to a draft until the analysis is complete?

Reply:
It was something I was originally planning on doing much later (probably one of the last audio things) just because I want to read and learn more about fourier analysis, plus this data isn't used in any meaningful way so I put it off to be much lower priority.

That being said, if it's something that you'd prefer being a higher priority, I can try and pick up fourier analysis sooner rather than later and focus on that.